python/ray/util/sgd/data/dataset.py

```diff
@@ -0,0 +1,127 @@
+from abc import ABC, abstractmethod
+
+import random
+import ray
+import torch
+
+from ray.util.sgd.data.utils import get_arr_partition
+
+
+@ray.remote
+def _load_remote(dataset, path):
+    return dataset.load(path)
+
+
+class Dataset(ABC, torch.utils.data.IterableDataset):
```

Comment:
Maybe we should change this to `FileDataset`? I think there's an opportunity here to make a more general `Dataset` class that doesn't necessarily have to load training data from a filepath.

Reply:
so path wasn't intended to strictly be a filepath (it could be an s3 path or hdfs path for example), but perhaps there's a better name...

Reply:
Right but this still assumes you're reading files from somewhere. What do you do if you have your training data in memory? I'm thinking we should also have a general `Dataset` that just takes in an iterable.